src/shared/user/user.service.js

```diff
@@ -0,0 +1,106 @@
+import User from './user.model';
+
+/**
+ * It will authenticate user password
+ *
+ * @param {Object} user
+ * @param {String} password
+ * @return {Boolean}
+ *
+ */
+function isPasswordCorrect(user, password) {
+    return user.authenticate(password);
+}
+
+/**
+ * It will find one user by query
+ *
+ * @param {Object} query
+ * @return {object}
+ *
+ */
+function findOneBy(query) {
+    return User.findOne(query);
+}
+
+/**
+ * It will return all users
+ * @return {Array}
+ *
+ */
+function getAllUsers() {
+    return User.find({}, '-salt -hashedPassword -__v');
+}
+
+/**
+ * It will return user by id
+ *
+ * @param {Number} userId
+ * @return {object}
+ *
+ */
+function getUserById(userId) {
+    return User.findById(userId);
+}
+
+/**
+ * It will create new user in database
+ *
+ * @param {object} body
+ * @return {object}
+ *
+ */
+function createUser(body) {
+    return User.create(body).then((result) => {
```

Comment:
calback inside .then is states for nothing, and it can be safely removed
```js
function createUser(body) {
    return User.create(body);
}
```
is the same code